tests/path_gateway_dag_test.go

```diff
@@ -563,10 +669,168 @@ func TestNativeDag(t *testing.T) {
+					dagJsonConvertedData = t
+					return true
+				}
+				return false
+			})),
+		},
+	}, specs.PathGatewayDAG)
+
+	if dagJsonConvertedData != nil {
+		rangeTests := SugarTests{
+			helpers.SingleRangeTestTransform(t, SugarTest{
+				Name: "Convert application/vnd.ipld.dag-cbor to application/vnd.ipld.dag-json with single range request includes correct bytes",
+				Hint: "",
+				Request: Request().
+					Path("/ipfs/{{cid}}/", dagCborCID).
+					Headers(
+						Header("Accept", "application/vnd.ipld.dag-json"),
+						Header("Range", "bytes=1-5"),
+					),
+				Response: Expect(),
+			}, helpers.SimpleByteRange(1, 5, dagJsonConvertedData[1:6]), dagJsonConvertedData),
+			helpers.MultiRangeTestTransform(t, SugarTest{
+				Name: "Convert application/vnd.ipld.dag-cbor to application/vnd.ipld.dag-json with multiple range request includes correct bytes",
+				Hint: "",
+				Request: Request().
+					Path("/ipfs/{{cid}}/", dagCborCID).
+					Headers(
+						Header("Accept", "application/vnd.ipld.dag-json"),
+						Header("Range", "bytes=1-5,93-104"),
+					),
+				Response: Expect(),
+			}, helpers.ByteRanges{
+				helpers.SimpleByteRange(1, 5, dagJsonConvertedData[1:6]),
+				helpers.SimpleByteRange(93, 104, dagJsonConvertedData[93:105])},
+				dagJsonConvertedData),
+		}
+
+		RunWithSpecs(t, rangeTests, specs.PathGatewayDAG)
+	}
+
+	var dagCborHTMLRendering []byte
+	RunWithSpecs(t, SugarTests{
+		SugarTest{
+			Name: "Convert application/vnd.ipld.dag-cbor to text/html",
+			Hint: "",
+			Request: Request().
+				Path("/ipfs/{{cid}}/", dagCborCID).
+				Headers(
+					Header("Accept", "text/html"),
+				),
+			Response: Expect().Body(Checks("", func(t []byte) bool {
+				innerCheck := Contains("</html>").Check(string(t))
+				if innerCheck.Success {
+					dagCborHTMLRendering = t
+					return true
+				}
+				return false
+			})),
+		},
+	}, specs.PathGatewayDAG)
+
+	if dagCborHTMLRendering != nil {
+		rangeTests := SugarTests{
+			helpers.SingleRangeTestTransform(t, SugarTest{
+				Name: "Convert application/vnd.ipld.dag-cbor to text/html with single range request includes correct bytes",
+				Hint: "",
+				Request: Request().
+					Path("/ipfs/{{cid}}/", dagCborCID).
+					Headers(
+						Header("Accept", "text/html"),
+						Header("Range", "bytes=1-5"),
+					),
+				Response: Expect(),
+			}, helpers.SimpleByteRange(1, 5, dagCborHTMLRendering[1:6]), dagCborHTMLRendering),
+			helpers.MultiRangeTestTransform(t, SugarTest{
+				Name: "Convert application/vnd.ipld.dag-cbor to text/html with multiple range request includes correct bytes",
+				Hint: "",
+				Request: Request().
+					Path("/ipfs/{{cid}}/", dagCborCID).
+					Headers(
+						Header("Accept", "text/html"),
+						Header("Range", "bytes=1-5,93-104"),
+					),
+				Response: Expect(),
+			}, helpers.ByteRanges{
+				helpers.SimpleByteRange(1, 5, dagCborHTMLRendering[1:6]),
+				helpers.SimpleByteRange(93, 104, dagCborHTMLRendering[93:105])},
+				dagCborHTMLRendering),
```

Comment:
Each of these test repeat the range parameters,

```go
...

						Header("Range", "bytes=1-5,93-104"),

...

				helpers.SimpleByteRange(1, 5, dagCborHTMLRendering[1:6]),
				helpers.SimpleByteRange(93, 104, dagCborHTMLRendering[93:105])},

...
```

Would that make sense to either, be explicit about generation:

```go
						Header("Range", "bytes=GENERATED"),
```

check that the ranges match?

Reply:
the passed header is in the request, the SimpleByteRange is for the response. Should be able to wrap them all up together (i.e. insist the request doesn't have the Range header) and auto-generate the header in the request using `request.Clone` though.

Reply:
🤦  my bad, I got confused between the generated part and the request, that way looks good.